server/app/controllers/admin/AdminProgramBlockPredicatesController.java

```diff
@@ -188,11 +254,196 @@ public Result updateVisibility(Request request, long programId, long blockDefini
+      BlockDefinition blockDefinition = programDefinition.getBlockDefinition(blockDefinitionId);
+
+      return ok(
+          predicatesConfigureView.renderEligibility(
+              request,
+              programDefinition,
+              blockDefinition,
+              getQuestionDefinitionsForForm(programDefinition, blockDefinition, form)));
+    } catch (ProgramNotFoundException | ProgramBlockDefinitionNotFoundException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  @Secure(authorizers = Authorizers.Labels.CIVIFORM_ADMIN)
+  public Result configureExistingEligibilityPredicate(
+      Request request, long programId, long blockDefinitionId) {
+    requestChecker.throwIfProgramNotDraft(programId);
+
+    try {
+      ProgramDefinition programDefinition = programService.getProgramDefinition(programId);
+      BlockDefinition blockDefinition = programDefinition.getBlockDefinition(blockDefinitionId);
+
+      ImmutableList<Long> questionIds =
+          blockDefinition
+              .eligibilityDefinition()
+              .orElseThrow(
+                  () ->
+                      new RuntimeException(
+                          String.format(
+                              "Block %d has no eligibility definition", blockDefinition.id())))
+              .predicate()
+              .getQuestions();
+
+      ImmutableList<QuestionDefinition> questionDefinitions =
+          getQuestionDefinitions(
+              programDefinition,
+              blockDefinition,
+              (QuestionDefinition questionDefinition) ->
+                  questionIds.contains(questionDefinition.getId()));
+
+      return ok(
+          predicatesConfigureView.renderEligibility(
+              request, programDefinition, blockDefinition, questionDefinitions));
+    } catch (ProgramNotFoundException | ProgramBlockDefinitionNotFoundException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  private ImmutableList<QuestionDefinition> getQuestionDefinitionsForForm(
+      ProgramDefinition programDefinition, BlockDefinition blockDefinition, DynamicForm form) {
+    return getQuestionDefinitions(
+        programDefinition,
+        blockDefinition,
+        (QuestionDefinition questionDefinition) ->
+            form.rawData().containsKey(String.format("question-%d", questionDefinition.getId())));
+  }
+
+  private ImmutableList<QuestionDefinition> getQuestionDefinitions(
+      ProgramDefinition programDefinition,
+      BlockDefinition blockDefinition,
+      Predicate<QuestionDefinition> predicate) {
+
+    try {
+      return programDefinition
+          .getAvailablePredicateQuestionDefinitions(blockDefinition.id())
+          .stream()
+          .filter(predicate)
+          .collect(ImmutableList.toImmutableList());
+    } catch (ProgramBlockDefinitionNotFoundException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  private ImmutableList<QuestionDefinition> getQuestionDefinitions(
```

Comment:
This is redundant with `getQuestionDefinitionsForForm` I believe

Reply:
good catch, done